core/src/main/java/org/apache/accumulo/core/clientImpl/bulk/BulkImport.java

```diff
@@ -149,6 +150,11 @@ public void load()
       if (mappings.isEmpty())
         throw new IllegalArgumentException("Attempted to import zero files from " + srcPath);
 
+      long tabletMaxSize = conf.getCount(Property.MASTER_BULK_MAX_TABLETS);
+      if (tabletMaxSize > 0 && mappings.keySet().size() > tabletMaxSize)
```

Comment:
This check is different than the one in #1560, which ensured a single file did not go to too many tablets.  This is checking that all files do not go to too many tablets.

Reply:
Yes that is correct.  I am not sure which is better but based on the original description in #1559, I took it as we want to prevent files from mapping to too many tablets.

Reply:
I guess the danger with this check is stopping ingest due to false positives?  Like if I have 100 tablets and 10 files going to all tablets could be normal. I would rather stop 1 file going to all 100 then 10 going to all 100.

Reply:
I just looked at #1559 and saw the following sentence.  I would be in favor of preventing a single file from going to too many tablets.  If someone has a large number of tablets and each file is going to a few tablets, they are probably ok with that like you said.

> It would be nice if we could establish a threshold in the bulk import process to abort when encountering a rfile that maps to more than a specified number of extents.